libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -118,7 +119,15 @@ private static Policy loadPluginPolicy(Path pluginRoot, boolean isModular, Strin
         // TODO: should this check actually be part of the parser?
```

Comment:
(Side note: we could have a separate validation function, but I would be inclined not to put this in the parser. It's easy to put _too much_ in a parser. Parse errors are best for cases where you _can't tell_ what the user is asking for, rather than cases where they asked for something invalid. In this case we know what they were asking for; it just makes no sense.)

Reply:
(An alternate view would be Data-Oriented Programming, which stipulates that invalid states should be unrepresentable. That would suggest that the validation should be done either in the parser or in the entitlement record constructors.)

(Also, side note: [Scope](https://github.com/elastic/elasticsearch/blob/ac0ba24847a6f09ef8686659cd16ca74bc1d79d1/libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/Scope.java#L19) isn't a record but probably should be.)